README.md

```diff
@@ -69,32 +69,67 @@ The project includes mock interactions of external APIs (GPC, SDS) implemented i
 
 The folder `docker/wiremock/stubs` describes the supported interactions.
 
-## Unit test section 
+## How to run tests
+
+**Warning**: Gradle uses a [Build Cache](https://docs.gradle.org/current/userguide/build_cache.html) to re-use compile and
+test outputs for faster builds. To re-run passing tests without making any code changes you must first run 
+`./gradle clean` to clear the build cache. Otherwise, gradle uses the cached outputs from a previous test execution to 
+pass the build.
 
 ### How to run unit tests:
-* Navigate to `service`
-* Run: `./gradlew test`
 
-### How to run all checks (unit, style etc):
-* `docker build --target=test`
+```shell script
+cd service
+./gradlew test
+```
+
+### How to run all checks:
+
+```shell script
+cd service
+./gradlew check
+```
 
-## How to run integration tests:
-* Navigate to `service`
-* Run: `./gradlew integrationTest`
+### How to run integration tests:
+
+```shell script
+cd service
+./gradlew integrationTest
+```
 
 Integration tests automatically start their external dependencies using [TestContainers](https://www.testcontainers.org/). 
 To disable this set the `DISABLE_TEST_CONTAINERS` environment variable to `true`.
 
-## How to run style check:
-* Navigate to `service`
-* Run: `./gradlew staticCodeAnalysis` 
+#### Example: Run integration tests with AWS S3 in-the-loop
+
+Use environment variables to configure the tests to use:
+* An actual S3 bucket
+* ActiveMQ running locally in Docker
+* MongoDB running locally in Docker
+
+1. Start activemq and mongodb dependencies manually
+
+    ```shell script
+    cd docker
+    docker-compose up -d activemq mongodb
+    ```
+
+2. Configure environment variables
+
+    If you're NOT running the test from an AWS instance with an instance role to access the bucket then the variables 
+    `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_REGION` must also be set.
+    
+    ```shell script
+    export DISABLE_TEST_CONTAINERS=true
+    export GP2GP_STORAGE_CONTAINER_NAME=your-s3-bucket-name
+    ```
 
-## How to run all checks:
-* Navigate to `service`
-* Run: `./gradlew check`
+3. Run the integration tests
 
-## How to run e2e tests:
-* `docker-compose -f docker-compose-integration-tests.yml build && docker-compose -f docker-compose-integration-tests.yml up --exit-code-from integration_tests`
+    ```shell script
+    cd ../service
```

Comment:
./service?

Reply:
The users is already in `docker/` if they followed the previous step